### ISSUE TITLE
ecmaFeatures.modules conflicts with block-scoped-var

### ISSUE THREAD

**Issue by gmathieu** [eslint/eslint#2967]
When `ecmaFeatures.module=true`, functions calls before declaration throws a `block-scoped-var` error. This test is using the default parser `v0.24.1` but I believe I've seen something similar using `babel-eslint`.

```javascript
(function () {
    fnDeclaredBelow(); // error: fnDeclaredBelow used outside of binding context
})();

function fnDeclaredBelow () {
    return true;
}
```

Here's the associated `.eslintrc`. If you switch module to `false` the above code validates properly.

```json
{
  "env": {
    "es6": true,
  },
  "rules": {
    "block-scoped-var": 2,
    "no-use-before-define": [2, "nofunc"],
    "strict": [2, "never"],
  },
  "ecmaFeatures": {
    "modules": true,
  },
}
```

**Comment by gyandeeps** [comment]
@eslint/eslint-team input please.

**Comment by nzakas** [comment]
@gmathieu what about if modules is false? Is the behavior different?

**Comment by mysticatea** [comment]
I confirmed this issue, and #3126 fixes this.

## test.js

```js
(function () {
    fnDeclaredBelow(); // error: fnDeclaredBelow used outside of binding context
})();

function fnDeclaredBelow () {
    return true;
}
```

## .eslintrc

```
ecmaFeatures:
    modules: true
rules:
    block-scoped-var: 2
```

## Command & Actual

```
> eslint -v
v1.0.0-rc-3
> eslint test.js

test.js
  3:5  error  "fnDeclaredBelow" used outside of binding context  block-scoped-var

✖ 1 problem (1 error, 0 warnings)
```

```
> git log -1
commit 6a77f79f507fddb574a438e9ab29118caeb4d7a8
Author: Toru Nagashima <star.ctor@gmail.com>
Date:   Wed Jul 22 22:38:06 2015 +0900

    Fix: `block-scoped-var` issues (fixes #2253, fixes #2747)
> node bin/eslint.js test.js
>
```

No errors.

## Expected

No errors.

**Comment by nzakas** [comment]
@mysticatea can you add a test case for this in #3126?

**Comment by mysticatea** [comment]
@nzakas Sure.
